src/logic/check.utils.ts

```diff
@@ -0,0 +1,108 @@
+import { ProviderJson, SuperJson } from '@superfaceai/one-sdk';
+import { join as joinPath } from 'path';
+
+import { EXTENSIONS } from '../common';
+import { exists, readdir, readFile } from '../common/io';
+
+export async function findLocalProfileSource(
+  superJson: SuperJson,
+  profile: {
+    name: string;
+    scope?: string;
+    version?: string;
+  }
+): Promise<string | undefined> {
+  //Check file property
+  const profileName = profile.scope
+    ? `${profile.scope}/${profile.name}`
+    : profile.name;
+  const profileSettings = superJson.normalized.profiles[profileName];
+  if (profileSettings !== undefined && 'file' in profileSettings) {
+    const resolvedPath = superJson.resolvePath(profileSettings.file);
+    if (await exists(resolvedPath)) {
+      return readFile(resolvedPath, { encoding: 'utf-8' });
+    }
+  }
+
+  //try to look in the grid for source file
+  const basePath = profile.scope ? joinPath('grid', profile.scope) : 'grid';
+  if (profile.version) {
+    const path = joinPath(
+      basePath,
+      `${profile.name}@${profile.version}${EXTENSIONS.profile.source}`
+    );
+    const resolvedPath = new SuperJson().resolvePath(path);
+    if (await exists(resolvedPath)) {
+      return readFile(resolvedPath, { encoding: 'utf-8' });
+    }
+  } else {
+    //Look for any version
+    const scopePath = superJson.resolvePath(basePath);
+    if (await exists(scopePath)) {
+      //Get files in profile directory
+      const files = (await readdir(scopePath, { withFileTypes: true }))
+        .filter(dirent => dirent.isFile() || dirent.isSymbolicLink())
+        .map(dirent => dirent.name);
+      //Find files with similar name to profile and with .ast.json extension
+      const path = files.find(
+        f => f.includes(profile.name) && f.endsWith(EXTENSIONS.profile.source)
```

Comment:
This may return false positives:
```
scope/
  profile-foo@1.0.0.supr
  profile-foobar@1.0.0.supr
```

When looking for `profile-foo` we might get `profile-foobar` instead. Instead of `.includes` we should:
```
f.startsWith(`${profile.name}@`)
```